src/components/icon/icon.spec.js

```diff
@@ -514,29 +510,39 @@ describe('MdIcon service', function() {
     });
 
     describe('icon group is not found', function() {
-      it('should log Error', function() {
-        var msg;
-        try {
-          $mdIcon('notfound:someIcon')
-            .catch(function(error) {
-              msg = error.data;
-            });
+      it('should log Error and reject', inject(function($log) {
+        var errorMessage = 'Cannot GET notfoundgroup.svg';
+        var caughtRejection = false;
+
+        $mdIcon('notfound:someIcon')
+          .catch(function(error) {
+            expect(error.data).toBe(errorMessage);
+            caughtRejection = true;
+          });
 
-          $httpBackend.flush();
-        } finally {
-          expect(msg).toEqual('Cannot GET notfoundgroup.svg');
-        }
-      });
+        $httpBackend.flush();
+
+        expect(caughtRejection).toBe(true);
+        expect($log.warn.logs[0]).toEqual([errorMessage]);
+      }));
     });
 
     describe('icon is not found', function() {
-      it('should not throw Error', function() {
-        expect(function(){
-          $mdIcon('notfound');
+      it('should log Error and reject', inject(function($log) {
```

Comment:
> I believe we have a separate test below which tests this exact case.

I don't see any test below :stuck_out_tongue: 

That said, I don't think you need to explicitly test that something does not throw - if it throws the test will fail anyway.
The changes in this test only test the underlying behavior, I didn't change the behavior (in the corresponding commit), so there was no breaking change.

The question is what _is_ the desired behavior ? Previously, if the icon/group was not found, there was a "Possibly unhandled rejection" error (with v1.6) which the user would not be able to silence no matter what. With this PR, this error is will only be thrown if the user doesn't handle the rejection (i.e. not chain a `.catch()` handler to `$mdIcon('nofoundicon.svg')`).
If the desired behavior is that it should never raise a "Possibly unhandled rejection" error (even if the user never caught the rejection), then more modification is necessary.

If you ask me, I can't think of a good reason why I would want to silence a "Possibly unhandled rejection" for a non-existing icon, but there might be a usecase I am missing (related to icons).
If there are such usecases, you might want to make this configurable (either globally or per request - depending on what makes most sense to you). E.g. throw the error by default, but let the user specify that the `$mdIcon`-related unhandled rejection are OK and should be silenced.